backend/internal/application/init.go

```diff
@@ -16,44 +16,35 @@
  * under the License.
  */
 
-// Package services provides HTTP service implementations for various domain operations.
-//
-//nolint:dupl // ApplicationService has similar structure to GroupService but they serve different domains
-package services
+// Package application provides functionality for managing applications.
+package application
 
 import (
 	"net/http"
 
-	"github.com/asgardeo/thunder/internal/application/handler"
+	"github.com/asgardeo/thunder/internal/cert"
 	"github.com/asgardeo/thunder/internal/system/middleware"
 )
 
-// ApplicationService defines the service for handling application-related requests.
-type ApplicationService struct {
-	applicationHandler *handler.ApplicationHandler
+// Initialize initializes the application service and registers its routes.
+func Initialize(mux *http.ServeMux, certService cert.CertificateServiceInterface) ApplicationServiceInterface {
+	appStore := NewCachedBackedApplicationStore()
```

Comment:
Shouldn't this private
```suggestion
	appStore := newCachedBackedApplicationStore()
```